File: swagger-codegen/spring/src/main/java/io/swagger/model/DispatchJobCreate.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.springframework.validation.annotation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * DispatchJobCreate
 */
@Validated
@javax.annotation.Generated(value = "io.swagger.codegen.languages.SpringCodegen", date = "2019-03-01T04:20:34.312Z")

public class DispatchJobCreate   {
  @JsonProperty("destination_address")
  private String destinationAddress = null;

  @JsonProperty("destination_address_id")
  private Long destinationAddressId = null;

  @JsonProperty("destination_lat")
  private Double destinationLat = null;

  @JsonProperty("destination_lng")
  private Double destinationLng = null;

  @JsonProperty("destination_name")
  private String destinationName = null;

  @JsonProperty("notes")
  private String notes = null;

  @JsonProperty("scheduled_arrival_time_ms")
  private Long scheduledArrivalTimeMs = null;

  @JsonProperty("scheduled_departure_time_ms")
  private Long scheduledDepartureTimeMs = null;

  public DispatchJobCreate destinationAddress(String destinationAddress) {
    this.destinationAddress = destinationAddress;
    return this;
  }

  /**
   * The address of the job destination, as it would be recognized if provided to maps.google.com. Optional if a valid destination address ID is provided.
   * @return destinationAddress
  **/
  @ApiModelProperty(example = "123 Main St, Philadelphia, PA 19106", value = "The address of the job destination, as it would be recognized if provided to maps.google.com. Optional if a valid destination address ID is provided.")


  public String getDestinationAddress() {
    return destinationAddress;
  }

  public void setDestinationAddress(String destinationAddress) {
    this.destinationAddress = destinationAddress;
  }

  public DispatchJobCreate destinationAddressId(Long destinationAddressId) {
    this.destinationAddressId = destinationAddressId;
    return this;
  }

  /**
   * ID of the job destination associated with an address book entry. Optional if valid values are provided for destination address or latitude/longitude. If a valid destination address ID is provided, address/latitude/longitude will be used from the address book entry. Name of the address book entry will only be used if the destination name is not provided.
   * @return destinationAddressId
  **/
  @ApiModelProperty(example = "67890", value = "ID of the job destination associated with an address book entry. Optional if valid values are provided for destination address or latitude/longitude. If a valid destination address ID is provided, address/latitude/longitude will be used from the address book entry. Name of the address book entry will only be used if the destination name is not provided.")


  public Long getDestinationAddressId() {
    return destinationAddressId;
  }

  public void setDestinationAddressId(Long destinationAddressId) {
    this.destinationAddressId = destinationAddressId;
  }

  public DispatchJobCreate destinationLat(Double destinationLat) {
    this.destinationLat = destinationLat;
    return this;
  }

  /**
   * Latitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.
   * @return destinationLat
  **/
  @ApiModelProperty(example = "123.456", value = "Latitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.")


  public Double getDestinationLat() {
    return destinationLat;
  }

  public void setDestinationLat(Double destinationLat) {
    this.destinationLat = destinationLat;
  }

  public DispatchJobCreate destinationLng(Double destinationLng) {
    this.destinationLng = destinationLng;
    return this;
  }

  /**
   * Longitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.
   * @return destinationLng
  **/
  @ApiModelProperty(example = "37.459", value = "Longitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.")


  public Double getDestinationLng() {
    return destinationLng;
  }

  public void setDestinationLng(Double destinationLng) {
    this.destinationLng = destinationLng;
  }

  public DispatchJobCreate destinationName(String destinationName) {
    this.destinationName = destinationName;
    return this;
  }

  /**
   * The name of the job destination. If provided, it will take precedence over the name of the address book entry.
   * @return destinationName
  **/
  @ApiModelProperty(example = "ACME Inc. Philadelphia HQ", value = "The name of the job destination. If provided, it will take precedence over the name of the address book entry.")


  public String getDestinationName() {
    return destinationName;
  }

  public void setDestinationName(String destinationName) {
    this.destinationName = destinationName;
  }

  public DispatchJobCreate notes(String notes) {
    this.notes = notes;
    return this;
  }

  /**
   * Notes regarding the details of this job.
   * @return notes
  **/
  @ApiModelProperty(example = "Ensure crates are stacked no more than 3 high.", value = "Notes regarding the details of this job.")


  public String getNotes() {
    return notes;
  }

  public void setNotes(String notes) {
    this.notes = notes;
  }

  public DispatchJobCreate scheduledArrivalTimeMs(Long scheduledArrivalTimeMs) {
    this.scheduledArrivalTimeMs = scheduledArrivalTimeMs;
    return this;
  }

  /**
   * The time at which the assigned driver is scheduled to arrive at the job destination.
   * @return scheduledArrivalTimeMs
  **/
  @ApiModelProperty(example = "1462881998034", required = true, value = "The time at which the assigned driver is scheduled to arrive at the job destination.")
  @NotNull


  public Long getScheduledArrivalTimeMs() {
    return scheduledArrivalTimeMs;
  }

  public void setScheduledArrivalTimeMs(Long scheduledArrivalTimeMs) {
    this.scheduledArrivalTimeMs = scheduledArrivalTimeMs;
  }

  public DispatchJobCreate scheduledDepartureTimeMs(Long scheduledDepartureTimeMs) {
    this.scheduledDepartureTimeMs = scheduledDepartureTimeMs;
    return this;
  }

  /**
   * The time at which the assigned driver is scheduled to depart from the job destination.
   * @return scheduledDepartureTimeMs
  **/
  @ApiModelProperty(example = "1462881998034", value = "The time at which the assigned driver is scheduled to depart from the job destination.")


  public Long getScheduledDepartureTimeMs() {
    return scheduledDepartureTimeMs;
  }

  public void setScheduledDepartureTimeMs(Long scheduledDepartureTimeMs) {
    this.scheduledDepartureTimeMs = scheduledDepartureTimeMs;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DispatchJobCreate dispatchJobCreate = (DispatchJobCreate) o;
    return Objects.equals(this.destinationAddress, dispatchJobCreate.destinationAddress) &&
        Objects.equals(this.destinationAddressId, dispatchJobCreate.destinationAddressId) &&
        Objects.equals(this.destinationLat, dispatchJobCreate.destinationLat) &&
        Objects.equals(this.destinationLng, dispatchJobCreate.destinationLng) &&
        Objects.equals(this.destinationName, dispatchJobCreate.destinationName) &&
        Objects.equals(this.notes, dispatchJobCreate.notes) &&
        Objects.equals(this.scheduledArrivalTimeMs, dispatchJobCreate.scheduledArrivalTimeMs) &&
        Objects.equals(this.scheduledDepartureTimeMs, dispatchJobCreate.scheduledDepartureTimeMs);
  }

  @Override
  public int hashCode() {
    return Objects.hash(destinationAddress, destinationAddressId, destinationLat, destinationLng, destinationName, notes, scheduledArrivalTimeMs, scheduledDepartureTimeMs);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DispatchJobCreate {\n");
    
    sb.append("    destinationAddress: ").append(toIndentedString(destinationAddress)).append("\n");
    sb.append("    destinationAddressId: ").append(toIndentedString(destinationAddressId)).append("\n");
    sb.append("    destinationLat: ").append(toIndentedString(destinationLat)).append("\n");
    sb.append("    destinationLng: ").append(toIndentedString(destinationLng)).append("\n");
    sb.append("    destinationName: ").append(toIndentedString(destinationName)).append("\n");
    sb.append("    notes: ").append(toIndentedString(notes)).append("\n");
    sb.append("    scheduledArrivalTimeMs: ").append(toIndentedString(scheduledArrivalTimeMs)).append("\n");
    sb.append("    scheduledDepartureTimeMs: ").append(toIndentedString(scheduledDepartureTimeMs)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: openapi-generator/csharp-nancyfx/src/Org.OpenAPITools/Models/DispatchJobCreate.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools.v1.Models
{
    /// <summary>
    /// DispatchJobCreate
    /// </summary>
    public sealed class DispatchJobCreate:  IEquatable<DispatchJobCreate>
    { 
        /// <summary>
        /// The address of the job destination, as it would be recognized if provided to maps.google.com. Optional if a valid destination address ID is provided.
        /// </summary>
        public string DestinationAddress { get; private set; }

        /// <summary>
        /// ID of the job destination associated with an address book entry. Optional if valid values are provided for destination address or latitude/longitude. If a valid destination address ID is provided, address/latitude/longitude will be used from the address book entry. Name of the address book entry will only be used if the destination name is not provided.
        /// </summary>
        public long? DestinationAddressId { get; private set; }

        /// <summary>
        /// Latitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.
        /// </summary>
        public double? DestinationLat { get; private set; }

        /// <summary>
        /// Longitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.
        /// </summary>
        public double? DestinationLng { get; private set; }

        /// <summary>
        /// The name of the job destination. If provided, it will take precedence over the name of the address book entry.
        /// </summary>
        public string DestinationName { get; private set; }

        /// <summary>
        /// Notes regarding the details of this job.
        /// </summary>
        public string Notes { get; private set; }

        /// <summary>
        /// The time at which the assigned driver is scheduled to arrive at the job destination.
        /// </summary>
        public long? ScheduledArrivalTimeMs { get; private set; }

        /// <summary>
        /// The time at which the assigned driver is scheduled to depart from the job destination.
        /// </summary>
        public long? ScheduledDepartureTimeMs { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use DispatchJobCreate.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public DispatchJobCreate()
        {
        }

        private DispatchJobCreate(string DestinationAddress, long? DestinationAddressId, double? DestinationLat, double? DestinationLng, string DestinationName, string Notes, long? ScheduledArrivalTimeMs, long? ScheduledDepartureTimeMs)
        {
            
            this.DestinationAddress = DestinationAddress;
            
            this.DestinationAddressId = DestinationAddressId;
            
            this.DestinationLat = DestinationLat;
            
            this.DestinationLng = DestinationLng;
            
            this.DestinationName = DestinationName;
            
            this.Notes = Notes;
            
            this.ScheduledArrivalTimeMs = ScheduledArrivalTimeMs;
            
            this.ScheduledDepartureTimeMs = ScheduledDepartureTimeMs;
            
        }

        /// <summary>
        /// Returns builder of DispatchJobCreate.
        /// </summary>
        /// <returns>DispatchJobCreateBuilder</returns>
        public static DispatchJobCreateBuilder Builder()
        {
            return new DispatchJobCreateBuilder();
        }

        /// <summary>
        /// Returns DispatchJobCreateBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>DispatchJobCreateBuilder</returns>
        public DispatchJobCreateBuilder With()
        {
            return Builder()
                .DestinationAddress(DestinationAddress)
                .DestinationAddressId(DestinationAddressId)
                .DestinationLat(DestinationLat)
                .DestinationLng(DestinationLng)
                .DestinationName(DestinationName)
                .Notes(Notes)
                .ScheduledArrivalTimeMs(ScheduledArrivalTimeMs)
                .ScheduledDepartureTimeMs(ScheduledDepartureTimeMs);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(DispatchJobCreate other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (DispatchJobCreate.
        /// </summary>
        /// <param name="left">Compared (DispatchJobCreate</param>
        /// <param name="right">Compared (DispatchJobCreate</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (DispatchJobCreate left, DispatchJobCreate right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (DispatchJobCreate.
        /// </summary>
        /// <param name="left">Compared (DispatchJobCreate</param>
        /// <param name="right">Compared (DispatchJobCreate</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (DispatchJobCreate left, DispatchJobCreate right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of DispatchJobCreate.
        /// </summary>
        public sealed class DispatchJobCreateBuilder
        {
            private string _DestinationAddress;
            private long? _DestinationAddressId;
            private double? _DestinationLat;
            private double? _DestinationLng;
            private string _DestinationName;
            private string _Notes;
            private long? _ScheduledArrivalTimeMs;
            private long? _ScheduledDepartureTimeMs;

            internal DispatchJobCreateBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.DestinationAddress property.
            /// </summary>
            /// <param name="value">The address of the job destination, as it would be recognized if provided to maps.google.com. Optional if a valid destination address ID is provided.</param>
            public DispatchJobCreateBuilder DestinationAddress(string value)
            {
                _DestinationAddress = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.DestinationAddressId property.
            /// </summary>
            /// <param name="value">ID of the job destination associated with an address book entry. Optional if valid values are provided for destination address or latitude/longitude. If a valid destination address ID is provided, address/latitude/longitude will be used from the address book entry. Name of the address book entry will only be used if the destination name is not provided.</param>
            public DispatchJobCreateBuilder DestinationAddressId(long? value)
            {
                _DestinationAddressId = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.DestinationLat property.
            /// </summary>
            /// <param name="value">Latitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.</param>
            public DispatchJobCreateBuilder DestinationLat(double? value)
            {
                _DestinationLat = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.DestinationLng property.
            /// </summary>
            /// <param name="value">Longitude of the destination in decimal degrees. Optional if a valid destination address ID is provided.</param>
            public DispatchJobCreateBuilder DestinationLng(double? value)
            {
                _DestinationLng = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.DestinationName property.
            /// </summary>
            /// <param name="value">The name of the job destination. If provided, it will take precedence over the name of the address book entry.</param>
            public DispatchJobCreateBuilder DestinationName(string value)
            {
                _DestinationName = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.Notes property.
            /// </summary>
            /// <param name="value">Notes regarding the details of this job.</param>
            public DispatchJobCreateBuilder Notes(string value)
            {
                _Notes = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.ScheduledArrivalTimeMs property.
            /// </summary>
            /// <param name="value">The time at which the assigned driver is scheduled to arrive at the job destination.</param>
            public DispatchJobCreateBuilder ScheduledArrivalTimeMs(long? value)
            {
                _ScheduledArrivalTimeMs = value;
                return this;
            }

            /// <summary>
            /// Sets value for DispatchJobCreate.ScheduledDepartureTimeMs property.
            /// </summary>
            /// <param name="value">The time at which the assigned driver is scheduled to depart from the job destination.</param>
            public DispatchJobCreateBuilder ScheduledDepartureTimeMs(long? value)
            {
                _ScheduledDepartureTimeMs = value;
                return this;
            }


            /// <summary>
            /// Builds instance of DispatchJobCreate.
            /// </summary>
            /// <returns>DispatchJobCreate</returns>
            public DispatchJobCreate Build()
            {
                Validate();
                return new DispatchJobCreate(
                    DestinationAddress: _DestinationAddress,
                    DestinationAddressId: _DestinationAddressId,
                    DestinationLat: _DestinationLat,
                    DestinationLng: _DestinationLng,
                    DestinationName: _DestinationName,
                    Notes: _Notes,
                    ScheduledArrivalTimeMs: _ScheduledArrivalTimeMs,
                    ScheduledDepartureTimeMs: _ScheduledDepartureTimeMs
                );
            }

            private void Validate()
            { 
                if (_ScheduledArrivalTimeMs == null)
                {
                    throw new ArgumentException("ScheduledArrivalTimeMs is a required property for DispatchJobCreate and cannot be null");
                } 
            }
        }

        
    }
}